src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/General/PreferencesType.php

```diff
@@ -48,11 +48,8 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $configuration = $this->getConfiguration();
         $isSslEnabled = $configuration->getBoolean('PS_SSL_ENABLED');
 
-        if ($isSslEnabled) {
```

Comment:
I guess we just had to change the `$isSslEnabled` to check if the user is in an HTTPS context.

The information does not come from the configuration but from the request, doesn't it?

Reply:
The objective of this check is to avoid merchant to enable it by mistake.

If he is not on this with a secure connection, he should not see the switch and the link should be visible instead.

Reply:
this is what the user wants in the forge issue: as I said, the problem is not technical but functional :+1: